spectral_cube/tests/test_dask.py

```diff
@@ -239,6 +241,15 @@ def func(x, add=None):
                                    rslt3.filled_data[:].value)
 
 
+def test_cube_on_cube(data_adv):
+    # regression test for #782
+    cube = DaskSpectralCube.read(data_adv)
```

Comment:
```suggestion
def test_cube_on_cube(data_vda_beams):
    # regression test for #782
    cube = DaskSpectralCube.read(data_vda_beams)
```
Did that test really catch the regression? When I ran it on HEAD, it passed, since `DaskSpectralCube.read(data_adv)` is returning a subclass of `SpectralCube`.
Above is returning a `DaskVaryingResolutionSpectralCube`, which is not, so it failed on HEAD, but passes with this PR.

Reply:
You're right!  This is specific to CASA .image files, so I misidentified the error.